gettsim/transfers/kinderzuschl/kinderzuschl.py

```diff
@@ -49,97 +44,89 @@ def kinderzuschl_m_hh(
     -------
 
     """
-    out = _kinderzuschl_nach_vermög_check_m_hh.copy()
-    cond = (~kinderzuschl_vorrang_hh & ~wohngeld_kinderzuschl_vorrang_hh) | (
+    if ((not kinderzuschl_vorrang_hh) & (not wohngeld_kinderzuschl_vorrang_hh)) | (
         anz_rentner_hh > 0
-    )
-    out.loc[cond] = 0
+    ):
+        out = 0.0
+    else:
+        out = _kinderzuschl_nach_vermög_check_m_hh
+
     return out
 
 
-def _kinderzuschl_vor_vermög_check_m_hh(
-    kinderzuschl_vorläufig_m: FloatSeries, hh_id: IntSeries
-) -> FloatSeries:
-    """Aggregate preliminary child benefit on household level.
+def kinderzuschl_vorläufig_m_hh_ab_07_2019(
+    arbeitsl_geld_2_brutto_eink_m_hh: float,
+    kinderzuschl_eink_min_m_hh: float,
+    kinderzuschl_kindereink_abzug_m_hh: float,
+    kinderzuschl_eink_anrechn_m_hh: float,
+) -> float:
+    """Calculate preliminary child benefit since 07/2019.
 
     Parameters
     ----------
-    kinderzuschl_vorläufig_m
-        See :func:`kinderzuschl_vorläufig_m`.
     hh_id
         See basic input variable :ref:`hh_id <hh_id>`.
+    arbeitsl_geld_2_brutto_eink_m_hh
+        See :func:`arbeitsl_geld_2_brutto_eink_m_hh`.
+    kinderzuschl_eink_min_m_hh
+        See :func:`kinderzuschl_eink_min_m_hh`.
+    kinderzuschl_kindereink_abzug_m_hh
+        See :func:`kinderzuschl_kindereink_abzug_m_hh`.
+    kinderzuschl_eink_anrechn_m_hh
+        See :func:`kinderzuschl_eink_anrechn_m_hh`.
 
     Returns
     -------
 
     """
-    return kinderzuschl_vorläufig_m.groupby(hh_id).max()
+    if arbeitsl_geld_2_brutto_eink_m_hh >= kinderzuschl_eink_min_m_hh:
+        out = max(
+            kinderzuschl_kindereink_abzug_m_hh - kinderzuschl_eink_anrechn_m_hh, 0.0
+        )
+    else:
+        out = 0.0
 
+    return out
 
-def kinderzuschl_vorläufig_m_ab_07_2019(
-    hh_id: IntSeries,
-    arbeitsl_geld_2_brutto_eink_m_hh: FloatSeries,
-    kinderzuschl_eink_min_m: FloatSeries,
-    kinderzuschl_kindereink_abzug_m: FloatSeries,
-    kinderzuschl_eink_anrechn_m: FloatSeries,
-) -> FloatSeries:
-    """Calculate preliminary child benefit since 07/2019.
+
+def kinderzuschl_vorläufig_m_hh_bis_06_2019(
+    arbeitsl_geld_2_brutto_eink_m_hh: float,
+    kinderzuschl_eink_min_m_hh: float,
+    kinderzuschl_eink_max_m_hh: float,
+    arbeitsl_geld_2_eink_m_hh: float,
+    kinderzuschl_kindereink_abzug_m_hh: float,
+    kinderzuschl_eink_anrechn_m_hh: float,
+) -> float:
+    """Calculate preliminary child benefit since 2005 until 06/2019.
```

Comment:
```suggestion
    """Calculate Kinderzuschlag since 2005 until 06/2019. Whether Kinderzuschlag or Arbeitslosengeld 2 applies will be checked later.
```